src/BlockEntities/MobSpawnerEntity.cpp

```diff
@@ -191,39 +199,22 @@ void cMobSpawnerEntity::SpawnEntity(void)
 
 int cMobSpawnerEntity::GetNearbyPlayersNum(void)
 {
-	auto SpawnerPos = Vector3d(0.5, 0.5, 0.5) + m_Pos;
 	int NumPlayers = 0;
 
-	class cCallback : public cChunkDataCallback
+	auto Callback = [&] (cEntity & a_Entity)
 	{
-	public:
-		cCallback(Vector3d a_SpawnerPos, int & a_NumPlayers) :
-			m_SpawnerPos(a_SpawnerPos),
-			m_NumPlayers(a_NumPlayers)
+		if (!a_Entity.IsPlayer())
 		{
+			return false;
 		}
-
-		virtual void Entity(cEntity * a_Entity) override
+		if ((m_Pos - a_Entity.GetPosition()).Length() <= m_RequiredPlayerRange)
 		{
-			if (!a_Entity->IsPlayer())
-			{
-				return;
-			}
-
-			if ((m_SpawnerPos - a_Entity->GetPosition()).Length() <= 16)
-			{
-				m_NumPlayers++;
-			}
+			NumPlayers++;
 		}
+		return false;
+	};
```

Comment:
Havin to return false on success is realy weird

Reply:
It's the C convention, adapted to a boolean return value  - 0 return values mean success, and 0 is false.

Reply:
I was confused because I only used function which only handle one entity and there it didn't matter if you returned true or false. But yeah makes sense